app/controllers/workflow_credential_controller.rb

```diff
@@ -0,0 +1,122 @@
+  before_action :check_prototype
+  before_action :check_privileges
+  before_action :get_session_data
+
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  include Mixins::GenericListMixin
+  include Mixins::GenericSessionMixin
+  include Mixins::GenericShowMixin
+  include Mixins::BreadcrumbsMixin
+
+  menu_section :workflow_credentials
+
+  def self.display_methods
+    %w[repositories]
+  end
+
+  def self.model
+    ManageIQ::Providers::Workflows::AutomationManager::Credential
+  end
+
+  def display_repositories
+    nested_list(ManageIQ::Providers::Workflows::AutomationManager::ConfigurationScriptSource)
+  end
+
+  def show_searchbar?
+    true
+  end
+
+  def button
+    case params[:pressed]
+    when 'embedded_automation_manager_credentials_add'
+      javascript_redirect(:action => 'new')
+    when 'embedded_automation_manager_credentials_edit'
+      javascript_redirect(:action => 'edit', :id => params[:miq_grid_checks])
+    when 'ansible_credential_tag'
+      tag(self.class.model)
+    when "workflow_repository_tag" # repositories from nested list
+      tag(ManageIQ::Providers::Workflows::AutomationManager::ConfigurationScriptSource)
+    end
+  end
+
+  def new
+    assert_privileges('embedded_automation_manager_credentials_add')
+    drop_breadcrumb(:name => _("Add a new Credential"), :url => "/workflow_credential/new")
+    @in_a_form = true
+    @id = 'new'
+  end
+
+  def edit
+    assert_privileges('embedded_automation_manager_credentials_edit')
+    auth = ManageIQ::Providers::Workflows::AutomationManager::Authentication.find(params[:id].to_i)
+    drop_breadcrumb(:name => _("Edit a Credential \"%{name}\"") % {:name => auth.name},
+                    :url  => "/workflow_credential/edit/#{params[:id]}")
+    @in_a_form = true
+    @id = auth.id
+  end
+
+  def toolbar
+    return 'workflow_repositories_center' if %w[repositories].include?(@display) # for nested list screen
+
+    %w[show_list].include?(@lastaction) ? 'workflow_credentials_center' : 'workflow_credential_center'
+  end
+
+  def download_data
+    assert_privileges('embedded_automation_manager_credentials_view')
+
+    super
+  end
+
+  def download_summary_pdf
+    assert_privileges('embedded_automation_manager_credentials_view')
+
+    super
+  end
+
+  def show
+    assert_privileges('embedded_automation_manager_credentials_view')
+
+    super
+  end
+
+  def show_list
+    assert_privileges('embedded_automation_manager_credentials_view')
+
+    super
+  end
+
+  def tag_edit_form_field_changed
+    assert_privileges('ansible_credential_tag')
+
+    $log.warn(caller.pretty_inspect)
+
+    super
+  end
+
+  private
+
+  def check_prototype
+    return if Settings.prototype.ems_workflows.enabled
+
+    log_privileges(false, "Workflows are not enabled. The user is not authorized for this task or item.")
+    raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.')
+  end
```

Comment:
This looks duplicated by the `WorkflowController` and `WorkflowRepositoryController`, could we move this to a mixin?

Reply:
NOTE this can be done in a separate PR if you prefer

Reply:
too late